filebeat/crawler/registrar.go

```diff
@@ -212,28 +199,15 @@ func (r *Registrar) getPreviousFile(newFilePath string, newFileInfo os.FileInfo)
 }
 
 func (r *Registrar) setState(path string, state *FileState) {
-	r.stateMutex.Lock()
-	defer r.stateMutex.Unlock()
 	r.state[path] = state
 }
 
 func (r *Registrar) getState(path string) (*FileState, bool) {
-	r.stateMutex.Lock()
-	defer r.stateMutex.Unlock()
 	state, exist := r.state[path]
 	return state, exist
 }
 
-func (r *Registrar) updateStateSource(state *FileState, path string) {
-	r.stateMutex.Lock()
-	defer r.stateMutex.Unlock()
-	state.Source = path
-}
-
 func (r *Registrar) getStateCopy() map[string]FileState {
-	r.stateMutex.Lock()
-	defer r.stateMutex.Unlock()
-
 	copy := make(map[string]FileState)
 	for k, v := range r.state {
 		copy[k] = *v
```

Comment:
Since `FileState` contains a `*FileStateOS`, having a copy does not guarantee that the immutability of the underlying data. `FileState.FileStateOS` should not be a pointer if you want to have a copy of the data.